packages/shopify-api/lib/webhooks/validate.ts

```diff
@@ -39,43 +39,24 @@ export function validateFactory(config: ConfigInterface) {
     const request: NormalizedRequest =
       await abstractConvertRequest(adapterArgs);
 
-    const log = logger(config);
+    const validHmacResult = await validateHmacRequest({
+      rawBody,
+      config,
+      type: ValidationType.Webhook,
+      ...adapterArgs,
+    });
 
-    const webhookCheck = checkWebhookRequest(rawBody, request.headers);
-    if (!webhookCheck.valid) {
-      await log.debug('Received malformed webhook request', webhookCheck);
-      return webhookCheck;
+    if (!validHmacResult.valid) {
+      return validHmacResult;
     }
 
-    const {hmac, valid: _valid, ...loggingContext} = webhookCheck;
-    await log.debug('Webhook request is well formed', loggingContext);
-
-    if (await validateHmacString(config, rawBody, hmac, HashFormat.Base64)) {
-      await log.debug('Webhook request is valid', loggingContext);
-      return webhookCheck;
-    } else {
-      await log.debug(
```

Comment:
The one thing about this refactor is we loose this specific logging about manually triggering webhooks. 
Maybe we should make some sort of override of the error messages for the general function?

Reply:
I added back in this specific logging for the webhook validation.